tensorflow/compiler/mlir/tfrt/transforms/lhlo_gpu_to_tfrt_gpu/gemm_pattern.cc

```diff
@@ -145,22 +162,19 @@ FailureOr<Value> CreateTfrtOps(
       loc, builder.getI32Type(), k_val);
 
   auto const_alpha =
-      MakeScalingFactorConstant(builder, loc, element_type, alpha);
+      MakeScalingFactorConstant(builder, loc, mlir_compute_type, alpha);
```

Comment:
Should this be using element_type instead of mlir_compute_type? If not, do we need the FP16 case handling in MakeScalingFactorConstant(), since mlir_compute_type has been adjusted to FP32 (by ComputationTypeFromElementType())?

Same for const_beta.

Reply:
This is a cuBLAS constraint: `alpha` and `beta` must use `compute_type`
https://docs.nvidia.com/cuda/cublas/index.html#cublas-GemmStridedBatchedEx

The fp16 handling in `MakeScalingFactorConstant` could be indeed removed, good catch.